astroquery/oac/core.py

```diff
@@ -0,0 +1,602 @@
+            parameters as a dict. The actual HTTP request is not made.
+            The default value is False.
+
+        Returns
+        -------
+        response : `requests.Response`
+            The HTTP response returned from the service.
+            All async methods should return the raw HTTP response.
+
+        Examples
+        --------
+        Searches can be done as a cone or a box. We first establish coordinates
+        and search parameters:
+
+        >>> import astropy.coordinates as coord
+        >>> import astropy.units as u
+        >>> from astroquery.oac import OAC
+        >>> # Sample coordinates. We are using GW170817.
+        >>> ra = 197.45037
+        >>> dec = -23.38148
+        >>> test_coords = coord.SkyCoord(ra=ra, dec=dec, unit=(u.deg, u.deg))
+        >>> test_radius = 10*u.arcsec
+        >>> test_height = 10*u.arcsec
+        >>> test_width = 10*u.arcsec
+
+        An example cone search:
+        >>> photometry = OAC.query_region(coordinates=test_coords,
+        ...                               radius=test_radius,
+        ...                               quantity="photometry",
+        ...                               attribute=["time", "magnitude",
+        ...                                          "e_magnitude", "band",
+        ...                                          "instrument"])
+
+        An example box search:
+        >>> photometry = OAC.query_region(coordinates=test_coords,
+        ...                               width=test_width, height=test_height,
+        ...                               quantity="photometry",
+        ...                               attribute=["time", "magnitude",
+        ...                                          "e_magnitude", "band",
+        ...                                          "instrument"])
+
+        These searches can be refined using the quantities, attributes, and
+        arguments, as with query_object.
+
+        """
+        # Default object name used for coordinate-based queries
+        event = 'catalog'
+
+        request_payload = self._args_to_payload(event,
+                                                quantity,
+                                                attribute,
+                                                argument,
+                                                data_format)
+
+        # Check that coordinate object is a valid astropy coordinate object
+        # Criteria/Code from ../sdss/core.py
+        if (not isinstance(coordinates, list) and
+            not isinstance(coordinates, Column) and
+            not (isinstance(coordinates, commons.CoordClasses) and
+                 not coordinates.isscalar)):
+
+            request_payload['ra'] = coordinates.ra.deg
+            request_payload['dec'] = coordinates.dec.deg
+
+        else:
+            try:
+                request_payload['ra'] = coordinates[0]
+                request_payload['dec'] = coordinates[1]
+            except Exception:
```

Comment:
It would be better to try to catch only specific exception types (e.g., `IndexError`).  General exception catching can interfere with the interpreter and produce some otherwise bad results.  If the code returns an unexpected Exception type, we should just raise that Exception and deal with it as a bug